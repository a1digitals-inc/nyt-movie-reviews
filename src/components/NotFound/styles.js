import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-style: oblique;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-top: 90px;
`;
