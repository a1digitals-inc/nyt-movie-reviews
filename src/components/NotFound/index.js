import React from 'react';
import PropTypes from 'prop-types';

import { Container, Content } from './styles';

const NotFound = ({ Code, Message }) => {
  return (
    <Container>
      <Content>
        <strong>{Code}</strong>
        <strong>{Message}</strong>
      </Content>
    </Container>
  );
};

NotFound.propTypes = {
  Code: PropTypes.number,
  Message: PropTypes.string.isRequired,
};

NotFound.defaultProps = {
  Code: null,
};

export default NotFound;
