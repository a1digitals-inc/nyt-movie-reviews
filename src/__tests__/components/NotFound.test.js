import React from 'react';
import { shallow } from 'enzyme';

import NotFound from '../../components/NotFound';

describe('NotFound component', () => {
  it('should render without crashing', () => {
    const props = {
      Code: 404,
      Message: 'Page not found',
    };
    const wrapper = shallow(<NotFound {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
