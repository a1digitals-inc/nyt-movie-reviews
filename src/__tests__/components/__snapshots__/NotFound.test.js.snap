// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render without crashing 1`] = `
<styled.div>
  <styled.div>
    <strong>
      404
    </strong>
    <strong>
      Page not found
    </strong>
  </styled.div>
</styled.div>
`;
